### PR TITLE
enable variable session time for mpcCoreKit

### PR DESCRIPTION
**Pull request by lwin-kyaw**
- provide `custom session time` for SessionToken Data based on the clients' subscribe plans
- create new test cases for `session time`

### CODE REVIEW

**Comment by lwin-kyaw** [comment]
This PR is depended on the torus.js PR, https://github.com/torusresearch/torus.js/pull/129

To make this `variable session time` works, we may need the following
- updates in torus-utils (torus.js) https://github.com/torusresearch/torus.js/pull/129
- need to include torus.js updates in the following dependencies
    1. CustomAuth (to include latest updates from torus.js)
    2. tKey ServiceProvider (to include CustomAuth dependency (torus.js) update)

*Will update the PR again once this, https://github.com/torusresearch/torus.js/pull/129, is merged and published*

**Comment by lwin-kyaw** [comment]
Updated `torus.js` as `torus.js@v12.2.0` has been released.

However, to make this variable session time works properly, we still need the following things
- Update `torus.js@v12.20` in `CustomAuth`. Deps update [PR](https://github.com/torusresearch/CustomAuth/pull/311)
- Update `@tkey/service-provider-torus` with `CustomAuth` updates

Currently, I've used `npm overrides` to update the peer dependencies in the package.json to make `session time` works.
```
"dependencies": {
...
},
"overrides": {
    "@toruslabs/customauth": {
    "@toruslabs/http-helpers": "^6.0.0",
    "@toruslabs/torus.js": "12.2.0"
  },
    "@tkey-mpc/service-provider-torus": {
    "@toruslabs/torus.js": "12.2.0"
  }
},
```

File: src/mpcCoreKit.ts [hunk 1]
```diff
@@ -116,6 +116,7 @@ export class Web3AuthMPCCoreKit implements ICoreKit {
     if (typeof options.manualSync !== "boolean") options.manualSync = false;
     if (!options.web3AuthNetwork) options.web3AuthNetwork = WEB3AUTH_NETWORK.MAINNET;
     if (!options.sessionTime) options.sessionTime = 86400;
+    if (!options.serverTimeOffset) options.serverTimeOffset = 0; // ms
```
Comment (himanshuchawla009) [inline src/mpcCoreKit.ts:119]: we need to pass this param in torus.js and customAuth constructor

**Review by himanshuchawla009 — CHANGES_REQUESTED** [review]

File: src/mpcCoreKit.ts (hunk 1 above)
Comment (lwin-kyaw) [reply src/mpcCoreKit.ts:119]: Made a PR, https://github.com/torusresearch/CustomAuth/pull/331 to include `serverTimeOffset` value in respective constructors.

Comment (himanshuchawla009) [reply src/mpcCoreKit.ts:119]: okay, till we merge and release custom auth, can you remove serverTimeOffset param from this sdk

**Review by himanshuchawla009 — APPROVED** [review]

**Pull request merged**